### ISSUE TITLE
Sketch 3.6 Issues

### ISSUE THREAD

**Issue by bomberstudios** [jamztang/MagicMirror#21]
Hi James,

we just found out that Magic Mirror is not working as expected on Sketch 3.6 Beta, and we thought we'd ping you : )

We get this when trying to run the Plugin:

```
23/02/2016 08:58:42.586	Sketch Internal[615]	MagicMirror: -[MagicMirror setImageQuality:] [Line 357] setImageQuality 0
23/02/2016 08:58:42.586	Sketch Internal[615]	MagicMirror: -[MMLayer setImageQuality:] [Line 195] setImageQuality 0
23/02/2016 08:58:42.586	Sketch Internal[615]	MagicMirror: -[MMLayer refresh] [Line 161] ratio: 1.24577861163227, quality: 0
23/02/2016 08:58:42.587	Sketch Internal[615]	+[MSExportRendererWithSVGSupport exporterForRequest:colorSpace:allowSubpixelAntialiasing:]: unrecognized selector sent to class 0x103c85c38
23/02/2016 08:58:42.587	Sketch Internal[615]	+[MSExportRendererWithSVGSupport exporterForRequest:colorSpace:allowSubpixelAntialiasing:]: unrecognized selector sent to class 0x103c85c38
23/02/2016 08:58:42.591	Sketch Internal[615]	(
	0   CoreFoundation                      0x00007fff8361dae2 __exceptionPreprocess + 178
	1   libobjc.A.dylib                     0x00007fff83a87f7e objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff83620a9d +[NSObject(NSObject) doesNotRecognizeSelector:] + 205
	3   CoreFoundation                      0x00007fff83559601 ___forwarding___ + 1009
	4   CoreFoundation                      0x00007fff83559188 _CF_forwarding_prep_0 + 120
	5   MagicMirror                         0x0000000115b60fc2 +[Sketch exportRendererForRequest:colorSpace:] + 152
	6   MagicMirror                         0x0000000115b59689 -[MMImageRenderer exportedImage] + 248
	7   MagicMirror                         0x0000000115b54f5b -[MMLayer mirrorWithArtboard:imageQuality:colorSpaceIdentifier:perspective:] + 239
	8   MagicMirror                         0x0000000115b5c206 -[MagicMirror mirrorLayer:fromArtboard:imageQuality:] + 147
	9   MagicMirror                         0x0000000115b552a2 -[MMLayer refresh] + 643
	10  MagicMirror                         0x0000000115b555fe -[MMLayer setImageQuality:] + 284
	11  MagicMirror                         0x0000000115b5c386 -[MagicMirror setImageQuality:forLayer:] + 104
	12  MagicMirror                         0x0000000115b5cc1a __31-[MagicMirror setImageQuality:]_block_invoke + 63
	13  CoreFoundation                      0x00007fff8354406d __53-[__NSArrayI enumerateObjectsWithOptions:usingBlock:]_block_invoke + 77
	14  CoreFoundation                      0x00007fff83543f39 -[__NSArrayI enumerateObjectsWithOptions:usingBlock:] + 201
	15  MagicMirror                         0x0000000115b5cb7b -[MagicMirror setImageQuality:] + 244
	16  MagicMirror                         0x0000000115b603b2 -[MMToolbarViewController comboBoxSelectionDidChange:] + 425
	17  MagicMirror                         0x0000000115b62967 -[MMComboBox comboBoxSelectionDidChange:] + 151
	18  CoreFoundation                      0x00007fff835d770c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	19  CoreFoundation                      0x00007fff835d767f ___CFXRegistrationPost_block_invoke + 63
	20  CoreFoundation                      0x00007fff835d6d47 _CFXRegistrationPost + 407
	21  CoreFoundation                      0x00007fff835d6ab2 ___CFXNotificationPost_block_invoke + 50
	22  CoreFoundation                      0x00007fff835d0d42 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1922
	23  CoreFoundation                      0x00007fff834bf145 _CFXNotificationPost + 693
	24  Foundation                          0x00007fff94294921 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
	25  CoreFoundation                      0x00007fff835d770c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	26  CoreFoundation                      0x00007fff835d767f ___CFXRegistrationPost_block_invoke + 63
	27  CoreFoundation                      0x00007fff835d6d47 _CFXRegistrationPost + 407
	28  CoreFoundation                      0x00007fff835d6ab2 ___CFXNotificationPost_block_invoke + 50
	29  CoreFoundation                      0x00007fff835d0d42 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1922
	30  CoreFoundation                      0x00007fff834bf145 _CFXNotificationPost + 693
	31  Foundation                          0x00007fff94294921 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
	32  CoreFoundation                      0x00007fff835d770c __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
	33  CoreFoundation                      0x00007fff835d767f ___CFXRegistrationPost_block_invoke + 63
	34  CoreFoundation                      0x00007fff835d6d47 _CFXRegistrationPost + 407
	35  CoreFoundation                      0x00007fff835d6ab2 ___CFXNotificationPost_block_invoke + 50
	36  CoreFoundation                      0x00007fff835d0d42 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1922
	37  CoreFoundation                      0x00007fff834bf145 _CFXNotificationPost + 693
	38  Foundation                          0x00007fff94294921 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
	39  AppKit                              0x00007fff80a6253c -[NSTableView _sendSelectionChangedNotificationForRows:columns:] + 175
	40  AppKit                              0x00007fff80a33723 -[NSTableView _enableSelectionPostingAndPost] + 424
	41  AppKit                              0x00007fff80bbba15 -[NSTableView mouseDown:] + 5358
	42  AppKit                              0x00007fff810bdd1d -[NSWindow _handleMouseDownEvent:isDelayedEvent:] + 6322
	43  AppKit                              0x00007fff810befad -[NSWindow _reallySendEvent:isDelayedEvent:] + 212
	44  AppKit                              0x00007fff80a17735 -[NSWindow sendEvent:] + 517
	45  AppKit                              0x00007fff80a13e49 -[NSApplication sendEvent:] + 2540
	46  AppKit                              0x00007fff80dc6945 -[NSComboBoxCell filterEvents:] + 888
	47  AppKit                              0x00007fff80dc80cc -[NSComboBoxCell popUp:] + 572
	48  libsystem_trace.dylib               0x00007fff8c3b307a _os_activity_initiate + 75
	49  AppKit                              0x00007fff80afe51d -[NSApplication sendAction:to:from:] + 460
	50  AppKit                              0x00007fff80b13486 -[NSControl sendAction:to:] + 86
	51  AppKit                              0x00007fff80b133b0 __26-[NSCell _sendActionFrom:]_block_invoke + 131
	52  libsystem_trace.dylib               0x00007fff8c3b307a _os_activity_initiate + 75
	53  AppKit                              0x00007fff80b1330d -[NSCell _sendActionFrom:] + 144
	54  AppKit                              0x00007fff80c132a6 __48-[NSCell trackMouse:inRect:ofView:untilMouseUp:]_block_invoke + 219
	55  libsystem_trace.dylib               0x00007fff8c3b307a _os_activity_initiate + 75
	56  AppKit                              0x00007fff80b113b0 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 1283
	57  AppKit                              0x00007fff80b66d58 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 744
	58  AppKit                              0x00007fff80dc6255 -[NSComboBoxCell trackMouse:inRect:ofView:untilMouseUp:] + 618
	59  AppKit                              0x00007fff80dc3d6e -[NSComboBox mouseDown:] + 366
	60  AppKit                              0x00007fff80a0f8dc forwardMethod + 126
	61  AppKit                              0x00007fff810bdd1d -[NSWindow _handleMouseDownEvent:isDelayedEvent:] + 6322
	62  AppKit                              0x00007fff810befad -[NSWindow _reallySendEvent:isDelayedEvent:] + 212
	63  AppKit                              0x00007fff80a17735 -[NSWindow sendEvent:] + 517
	64  AppKit                              0x00007fff80a13e49 -[NSApplication sendEvent:] + 2540
	65  AppKit                              0x00007fff8094703a -[NSApplication run] + 796
	66  AppKit                              0x00007fff808c9520 NSApplicationMain + 1176
	67  libdyld.dylib                       0x00007fff95c955ad start + 1
)
```

As you know, we've deprecated sub pixel antialiasing, so you'll need to use `[MSExportRendererWithSVGSupport exporterForRequest:colorSpace:]` instead of `[MSExportRendererWithSVGSupport exporterForRequest:colorSpace:allowSubpixelAntialiasing:]`

Hope it helps!

**Comment by jamztang** [comment]
Thanks for the heads up, will be working on it!

**Comment by jamztang** [comment]
Fixed on version 2.0.3, thanks!
